clients/src/main/java/org/apache/kafka/common/config/SslConfigs.java

```diff
@@ -69,7 +69,7 @@ public class SslConfigs {
 
     public static final String SSL_KEYSTORE_TYPE_CONFIG = "ssl.keystore.type";
     public static final String SSL_KEYSTORE_TYPE_DOC = "The file format of the key store file. "
-            + "This is optional for client.";
+            + "This is optional for client. The currently supported values for kafka are [JKS , PKCS12, PEM].";
```

Comment:
Our documentation generator has the ability to get the supported values from the validator. Here is an example:
```
.define(LogMessageTimestampTypeProp, STRING, Defaults.LogMessageTimestampType, in("CreateTime", "LogAppendTime"), MEDIUM, LogMessageTimestampTypeDoc)
```
Have you considered doing something similar here? Is there a reason not to do this?

Reply:
Hi @dajac  Thank you for your reply.
As I mentioned in JIRA:
> I'm not sure if it's appropriate to add a validator here. I just want to supplement the documentation.
Maybe users can customize their own SslEngineFactory through the `ssl.engine.factory.class` parameter to support other values than [JKS , PKCS12, PEM].

Now the documentation for these two parameters does not show any other optional values except for a default value of `JKS`. This can confuse users.

Reply:
In the default implementation class `DefaultSslEngineFactory`, the supported values for these two parameters are `[JKS , PKCS12, PEM]`

Reply:
If the accepted values depends on `ssl.engine.factory.class`, why would we want to add them to the documentation of `ssl.keystore.type`? This is misleading for users.

Reply:
From the existing documentation, the user does not know what other values to choose than JKS. Unless they read the source code or related [KIP-651.](https://cwiki.apache.org/confluence/display/KAFKA/KIP-651+-+Support+PEM+format+for+SSL+certificates+and+private+key)
I think adding a supplement to the documentation here to show what values are supported by default in Kafka can help users better.

Reply:
In this case, we should be clear about the fact that those are only accepted by the default `ssl.engine.factory.class`.

Reply:
ok i will improve the documentation later.